apps/group/contracts/Group.sol

```diff
@@ -45,26 +69,113 @@ contract Group is App, Initializable, IForwarder, EVMCallScriptRunner {
     */
     function removeMember(address _entity) auth(REMOVE_MEMBER_ROLE) external {
         require(isGroupMember(_entity));
-        members[_entity] = false;
+        isMember[_entity] = false;
+        for (uint i = 0; i < members.length; i++) {
+            if (members[i] == _entity) {
+                members[i] = members[members.length - 1];
+                delete members[members.length - 1];
+                members.length -= 1;
+                break;
+            }
+        }
+
+        if (required > members.length)
+            changeRequirement(members.length);
```

Comment:
Because `changeRequirement` could be called if a requirement reset is needed, in order for `removeMember` to work, the sender must also have the `CHANGE_REQUIREMENTS_ROLE ` role. I would say, in case `removeMember` requires a change in requirement, always allow it to happen. 

A good way to implement this could be moving the logic for changeRequirement to an internal function that both `removeMember` and external `changeRequirement` call.